modules/apps/frontend-js/frontend-js-bootstrap-support-web/src/main/resources/META-INF/resources/DropdownProvider.ts

```diff
@@ -117,18 +128,30 @@ class DropdownProvider {
 	};
 
 	_getMenu(trigger: any) {
-		return trigger.parentElement.querySelector('.dropdown-menu');
+		return trigger.parentElement.querySelector(Selector.MENU);
 	}
 
 	_getTrigger(menu: any) {
 		return menu.parentElement.querySelector('.dropdown-toggle');
 	}
 
-	_onKeyDown = (event: any) => {
+	_onItemKeyDown = (event: any) => {
+		const keyCode = event.keyCode;
+
+		if (keyCode !== KEYCODES.TAB && keyCode !== KEYCODES.ESCAPE) {
+			return;
+		}
+
+		this.hide({menu: event.delegateTarget.closest(Selector.MENU)});
```

Comment:
This is the key part that was not correct in the previous version. The previous version had `parentElement.parentElement` instead of `closest`, tying the solution to specific markup.